packages/did/src/DidResolver/DidResolver.ts

```diff
@@ -33,129 +28,98 @@ import { getFullDidUri, parseDidUri } from '../Did.utils.js'
-      const isFullDidDeleted = await queryDidDeletionStatus(did)
-      if (isFullDidDeleted) {
-        return {
-          // No canonicalId and no details are returned as we consider this DID deactivated/deleted.
-          metadata: {
-            deactivated: true,
-          },
-        }
-      }
-      // If no full DID details nor deletion info is found, the light DID is un-migrated.
-      // Metadata will simply contain `deactivated: false`.
-      return {
-        details,
-        metadata: {
-          deactivated: false,
-        },
-      }
+  }
+
+  if (type === 'full' && !fullDid) {
+    return null
+  }
+
+  const details = Did.parseDetailsFromLightDid(did, false)
+
+  // If a full DID with same subject is present, return the resolution metadata accordingly.
+  if (fullDid) {
+    return {
+      details,
+      metadata: {
+        canonicalId: getFullDidUri(did),
+        deactivated: false,
+      },
     }
-    default:
-      throw new SDKErrors.UnsupportedDidError(did)
+  }
+
+  // If no full DID details nor deletion info is found, the light DID is un-migrated.
+  // Metadata will simply contain `deactivated: false`.
+  return {
+    details,
+    metadata: {
+      deactivated: false,
+    },
   }
 }
 
 /**
  * Resolve a DID key URI to the key details.
  *
- * @param didUri The DID key URI.
+ * @param keyUri The DID key URI.
  * @returns The details associated with the key.
  */
 export async function resolveKey(
-  didUri: DidResourceUri
+  keyUri: DidResourceUri
 ): Promise<ResolvedDidKey | null> {
-  const { did, fragment: keyId, type } = parseDidUri(didUri)
+  const { did, fragment: keyId } = parseDidUri(keyUri)
 
   // A fragment (keyId) IS expected to resolve a key.
   if (!keyId) {
-    throw new SDKErrors.InvalidDidFormatError(didUri)
-  }
-
-  switch (type) {
-    case 'full': {
-      const details = await queryDetails(didUri)
-      const key = details && Did.getKey(details, keyId)
-      if (!key) {
-        return null
-      }
-      const { includedAt } = key
-      return {
-        controller: did,
-        id: didUri,
-        publicKey: key.publicKey,
-        type: key.type,
-        ...(includedAt && { includedAt }),
-      }
-    }
-    case 'light': {
-      const resolvedDetails = await resolve(didUri)
-      if (!resolvedDetails) {
-        throw new SDKErrors.InvalidDidFormatError(didUri)
-      }
-      if (!resolvedDetails.details) {
-        return null
-      }
-      const key = Did.getKey(resolvedDetails.details, keyId)
-      if (!key) {
-        return null
-      }
-      return {
-        controller: did,
-        id: didUri,
-        publicKey: key.publicKey,
-        type: key.type,
-      }
-    }
-    default:
-      throw new SDKErrors.UnsupportedDidError(didUri)
+    throw new SDKErrors.InvalidDidFormatError(keyUri)
+  }
+
+  const resolved = await resolve(keyUri)
```

Comment:
so the `resolve` function ignores the key fragment part of the key uri?
No matter if yes or no, I would still just resolve the `did` uri here, since you already got it above and represent more what you actually are  doing (resolving the did document, not just the key)

Reply:
Done in https://github.com/KILTprotocol/sdk-js/commit/a1257df987a4cf856c6e3ffe35673b75b2a2ff11